instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/CurrentNanoTime.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.concurrent.TimeUnit;
+import java.util.function.LongSupplier;
+
+/**
+ * This utility class allows to retrieve an {@link Instant} with the current time measured with
+ * nanosecond precision.
+ */
+public final class CurrentNanoTime {
```

Comment:
I think instead of this class defaults should return `-1` or `null` - we want to use the SDK's `Clock` as much as possible, only override as an exception. This is not quite as important for `start` but very improtant for `end` which should only be using nano time, not wall time

Reply:
awwww, we lose the fix for #2600, oh well.. 😄